### ISSUE TITLE
Memory leak with actix-web 2.0 and actix-rt 1.0.0 for echo websocket server

### ISSUE THREAD

**Issue by jeandeducla** [actix/actix-web#1523]
I took this actix-web example: [websocket echo server](https://github.com/actix/examples/tree/master/websocket), I just changed the client so it spawns `N` `ChatClient` that sends message every 1 seconds for `T` seconds (I pasted the client code below) and then close. 

I did the following test: I start the server, I spawn 200 sessions for 5min, wait some minutes, spawn another 200 sessions for 5min, wait... I expect the server memory to increase on the first 200 connections and then decrease to some threshold. On the next 200 connections I also expect the server memory to increase and then decrease to the same previous threshold. However that is not what I observe: the memory does not decrease once the connections close and it does not go back to the original threshold, it just keeps increasing. 

I ran this test using docker containers in kubernetes, and here is the container cpu and memory timeseries (the 200 sessions spawning correspond to the cpu bursts):
`cpu`
![image](https://user-images.githubusercontent.com/28356783/82465914-12524000-9ac0-11ea-8afd-89f3434a39c3.png)

`memory`
![image](https://user-images.githubusercontent.com/28356783/82465958-1da56b80-9ac0-11ea-9c91-84668d3490ad.png)

I observe the same thing when running this on my machine: the memory of the server process keeps increasing.

The versions used are the one from the example's `Cargo.toml` (I enforced actix-rt to `1.0.0` as suggested in this [issue](https://github.com/actix/actix-net/issues/129)), so:
```
[dependencies]
actix = "0.9.0"
actix-codec = "0.2.0"
actix-web = "2.0.0"
actix-web-actors = "2.0.0"
actix-files = "0.2.1"
actix-rt = "=1.0.0"
awc = "1.0.1"
env_logger = "0.7"
futures = "0.3.1"
bytes = "0.5.3"
```

## Expected Behavior
When the server handles connections, memory may increase and once the connections are all closed and the server is idle, memory should stay at some level that remains the same during the server lifetime (when the server is idle its memory should be around this threshold).

## Current Behavior
Server memory never decreases even when the server is idle.

## Steps to Reproduce
As I said I ran this in kubernetes, but you could see the same thing on your machine, I could not test on other platforms:
1. run the server and check the initial process memory
2. spawn 200 peers or more with the provided client
3. when the sessions close, check again the server process memory.
4. repeat

## Your Environment
* Rust Version (I.e, output of `rustc -V`): `rustc 1.41.1 (f3e1a954d 2020-02-24)`
* Actix Web Version: `2.0.0`
* My machine: `Linux castle1-3 5.6.6-arch1-1 #1 SMP PREEMPT Tue, 21 Apr 2020 10:35:16 +0000 x86_64 GNU/Linux`

`client.rs`:
```rust
//! Simple websocket client.
use std::time::Duration;

use actix::io::SinkWrite;
use actix::*;
use actix_codec::Framed;
use awc::{
    error::WsProtocolError,
    ws::{Codec, Frame, Message},
    BoxedSocket, Client,
};
use futures::future::FutureExt;
use futures::stream::{SplitSink, StreamExt};

fn main() {
    ::std::env::set_var("RUST_LOG", "actix_web=info");
    env_logger::init();

    let sys = System::new("websocket-client");

    for i in 0..200 {
        Arbiter::spawn(async move {
            ChatClient {
                msg: i.to_string(),
                writer: None,
                msg_frequency: Duration::from_secs(1),
                session_duration: Duration::from_secs(300),
            }
            .start();
        });
    }
    sys.run().unwrap();
}

type Writer = SinkWrite<Message, SplitSink<Framed<BoxedSocket, Codec>, Message>>;

pub struct ChatClient {
    pub writer: Option<Writer>,
    pub msg: String,
    pub msg_frequency: Duration,
    pub session_duration: Duration,
}

impl Actor for ChatClient {
    type Context = Context<Self>;

    fn started(&mut self, ctx: &mut Context<Self>) {
        self.connect(ctx);
        self.disconnect(ctx);
    }
}

impl ChatClient {
    fn connect(&self, ctx: &mut Context<Self>) {
        async move {
            let (response, framed) = Client::new()
                .ws("http://127.0.0.1:8080/ws")
                .connect()
                .await
                .map_err(|e| {
                    println!("Error: {}", e);
                })
                .unwrap();
            println!("Server response: {:?}", response);
            framed
        }
        .boxed_local()
        .into_actor(self)
        .then(|framed, act, ctx| {
            let (sink, stream) = framed.split();
            ChatClient::add_stream(stream, ctx);
            act.writer = Some(SinkWrite::new(sink, ctx));
            fut::ready(())
        })
        .wait(ctx);
    }

    fn send_message(&self, ctx: &mut Context<Self>) {
        let msg = self.msg.clone();
        ctx.run_later(self.msg_frequency, move |act, ctx| {
            act.write(Message::Text(msg));
            println!("msg send");
            act.send_message(ctx);
        });
    }

    fn disconnect(&self, ctx: &mut Context<Self>) {
        ctx.run_later(self.session_duration, |_, ctx| {
            println!("session closed");
            ctx.stop();
        });
    }

    fn write(&mut self, msg: Message) {
        if let Some(writer) = self.writer.as_mut() {
            if let Err(e) = writer.write(msg) {
                println!("Error sending message: {}", e);
            }
        }
    }
}

/// Handle server websocket messages
impl StreamHandler<Result<Frame, WsProtocolError>> for ChatClient {
    fn handle(&mut self, msg: Result<Frame, WsProtocolError>, _: &mut Context<Self>) {
        match msg {
            Ok(Frame::Text(txt)) => println!("Sever: {:?}", txt),
            Ok(Frame::Ping(msg)) => self.write(Message::Pong(msg)),
            _ => (),
        }
    }

    fn started(&mut self, ctx: &mut Self::Context) {
        self.send_message(ctx);
    }

    fn finished(&mut self, ctx: &mut Context<Self>) {
        println!("disconnected");
        self.connect(ctx);
    }
}

impl actix::io::WriteHandler<WsProtocolError> for ChatClient {}
```

**Comment by robjtede** [comment]
Thanks for the detailed report. Probably releated to #1428 but we'll keep both open until it's fixed.

**Comment by robjtede** [comment]
Tenatively fixed with #1580.
